File: finder4j-backtrack/src/test/java/com/github/k0zka/finder4j/backtrack/termination/TimeoutTerminationStrategyTest.java
package com.github.k0zka.finder4j.backtrack.termination;

import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import com.github.k0zka.finder4j.backtrack.State;
import com.github.k0zka.finder4j.backtrack.Step;

@RunWith(MockitoJUnitRunner.class)
public class TimeoutTerminationStrategyTest {

	@Mock
	State state;

	@Test
	public void stop() {
		Assert.assertTrue(new TimeoutTerminationStrategy<State, Step<State>>(
				System.currentTimeMillis() - 1000).stop(state));
		Assert.assertFalse(new TimeoutTerminationStrategy<State, Step<State>>(
				System.currentTimeMillis() + 1000).stop(state));
	}
}

File: finder4j-backtrack/src/test/kotlin/com/github/k0zka/finder4j/backtrack/termination/TimeoutTerminationStrategyTest.kt
package com.github.k0zka.finder4j.backtrack.termination

import com.nhaarman.mockito_kotlin.mock
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class TimeoutTerminationStrategyTest {

	private val state: Any = mock()

	@Test
	fun stop() {
		assertTrue(TimeoutTerminationStrategy<Any>(
				System.currentTimeMillis() - 1000).stop(state))
		assertFalse(TimeoutTerminationStrategy<Any>(
				System.currentTimeMillis() + 1000).stop(state))
	}
}
